src/pipecat/utils/asyncio.py

```diff
@@ -174,10 +174,19 @@ async def cancel_task(self, task: asyncio.Task, timeout: Optional[float] = None)
         except asyncio.CancelledError:
             # Here are sure the task is cancelled properly.
             pass
+        except GeneratorExit as e:
+            logger.exception(f"{name}: generator exit error while cancelling task: {e}")
+            raise
```

Comment:
`BaseException` should cover `GeneratorExit`. Maybe we can just use that since we print the exception.

Reply:
True. I initially had this one with the intention of maybe not raising the exception in this case. But then I changed my mind. Yeah, for logging only, BaseException should be enough. 👍

Reply:
Done.